src/dict/sounddir.cc

```diff
@@ -438,7 +439,32 @@ vector< sptr< Dictionary::Class > > makeDictionaries( Config::SoundDirs const &
 
     string indexFile = indicesDir + dictId;
 
-    if ( Dictionary::needToRebuildIndex( dictFiles, indexFile ) || indexIsOldOrBad( indexFile ) ) {
+    // Check if the soundDir and its subdirs' modification date changed, that means the user modified the sound files inside
+
+    bool soundDirModified = false;
+    {
+      QFileInfo info{};
+      info.setFile( QString::fromStdString( indexFile ) );
+
+      QDateTime indexFileModifyTime = info.lastModified();
+      info.setFile( dir.canonicalPath() );
+      QDateTime soundDirNewestModifyTime = info.lastModified();
+
+      QDirIterator it( dir, QDirIterator::Subdirectories );
+      while ( it.hasNext() ) {
+        info.setFile( it.next() );
+        auto thisDirModifiedTime = info.lastModified();
+        if ( thisDirModifiedTime > soundDirNewestModifyTime ) {
+          soundDirNewestModifyTime = thisDirModifiedTime;
+        }
+      }
+
+      if ( soundDirNewestModifyTime > indexFileModifyTime ) {
+        soundDirModified = true;
+      }
```

Comment:
move these lines into the above while loop ,

Reply:
Yes, you are right. I don't know how I missed such basics :sweat_smile:.

Fixed.

Reply:
I also just noticed that QDirIterator's first result is the dir itself, so there is no need to check separately.